workflow/controller/operator.go

```diff
@@ -169,6 +179,15 @@ func (woc *wfOperationCtx) operate() {
 
 	woc.log.Infof("Processing workflow")
 
+	// Loading WorkflowSpec for execution.
+	err := woc.loadWorkflowSpec()
+	if err != nil {
+		woc.log.Errorf("Unable to get Workflow Template Reference for workflow, %s error: %s", woc.wf.Name, err)
+		woc.markWorkflowError(err, true)
+		woc.persistUpdates()
```

Comment:
`persistUpdates()` is called as a `defer` function above, so by adding it here you'll be calling it twice and making two API calls to `Update`. This should be removed.
```suggestion
```